packages/passport-interface/test/SubscriptionManager.spec.ts

```diff
@@ -33,6 +34,67 @@ describe("Subscription Manager", async function () {
     MockDate.reset();
   });
 
+  function setupTestManager(
+    subs: { provider: string; feed: string }[]
+  ): FeedSubscriptionManager {
+    const manager = new FeedSubscriptionManager(new MockFeedApi());
+    for (const sub of subs) {
+      const providerUrl = "https://" + sub.provider;
+      manager.getOrAddProvider(providerUrl, sub.provider);
+      manager.subscribe(providerUrl, {
+        description: "description of " + sub.feed,
+        id: sub.feed,
+        name: "name of " + sub.feed,
+        permissions: [],
+        inputPCDType: undefined,
+        partialArgs: undefined,
+        credentialRequest: {
+          signatureType: "sempahore-signature-pcd"
+        }
+      });
+    }
+
+    expect(manager.getActiveSubscriptions()).to.have.length(subs.length);
+    for (const sub of subs) {
+      const providerUrl = "https://" + sub.provider;
+      expect(manager.hasProvider(providerUrl)).to.be.true;
+      expect(
+        manager.getSubscriptionsByProviderAndFeedId(providerUrl, sub.feed)
+      ).to.have.length(1);
+    }
+    return manager;
+  }
+
+  function expectAllTestSubs(
+    mgr: FeedSubscriptionManager,
+    subs: { provider: string; feed: string }[]
+  ): void {
+    for (const { provider, feed } of subs) {
+      const subs1 = mgr.getSubscriptionsByProviderAndFeedId(
+        "https://" + provider,
+        feed
+      );
+      expect(subs1).to.have.length(1);
+    }
+  }
+
+  function expectSameSubs(
+    mgr1: FeedSubscriptionManager,
+    mgr2: FeedSubscriptionManager
+  ): void {
+    expect(mgr1.getActiveSubscriptions().length).to.eq(
+      mgr2.getActiveSubscriptions().length
+    );
+    for (const sub1 of mgr1.getActiveSubscriptions()) {
+      const subs2 = mgr2.getSubscriptionsByProviderAndFeedId(
+        sub1.providerUrl,
+        sub1.feed.id
+      );
+      expect(subs2).to.have.length(1);
+      expect(subs2[0]).to.deep.eq(sub1);
+    }
+  }
+
```

Comment:
super nitpicky so feel free to ignore: these two functions feel like they are really tailored for your particular test suite, but feel like they have some underlying generality that may be worth teasing out. e.g. stuff like `[0]` generally scares me, and i'm not sure why `https://` needs to be prepended like that.

Reply:
Maybe an example of how I've done this sort of thing elsewhere may help to communicate what I mean: https://github.com/proofcarryingdata/zupass/blob/main/apps/passport-server/test/semaphore/checkSemaphore.ts#L17-L37

Reply:
expectAllTestSubs is definitely tailored to the test data generation in the previous function.  It's all about formatting the test data so it can be derived from very few inputs.  expectSameSubs isn't really test-case-specific in any way.
I'm not clear what if anything you're suggesting I do differently.